sherpa/utils/src/cephes/const.c

```diff
@@ -89,14 +89,16 @@ double LOGSQ2 =  3.46573590279972654709E-1;    /* log(2)/2 */
 double THPIO4 =  2.35619449019234492885;       /* 3*pi/4 */
 double TWOOPI =  6.36619772367581343075535E-1; /* 2/pi */
 #ifdef INFINITIES
-double INFINITY = strtod("INF");  /* 99e999; */
+#ifndef INFINITY
```

Comment:
The preprecessor conditionals here change the behavior compared to the original code due to the nesting. Is this intended?

in the original-  if INFINITIES is not defined, INFINITY is set to 1.79...

    #ifdef INFINITIES
        double INFINITY = strtod("INF");  /* 99e999; */
    #else
        double INFINITY =  1.79769313486231570815E308;    /* 2**1024*(1-MACHEP) */
    #endif

in the modified code- if INFINITIES is not defined, nothing happens...

    #ifdef INFINITIES
        #ifndef INFINITY
            double INFINITY = 1.0/0.0;
        #else
            double INFINITY =  1.79769313486231570815E308;    /* 2**1024*(1-MACHEP) */
        #endif
    #endif

Reply:
The following simple program will test if one can use ~1.79e308 and 0 for infinity and NaN, respectively; Moreover a check if 1.0/0.0 and 1.0/0. - 1/0.0 are legit definitions for infinity and NaN:

```
devel12-107: cat tst_inf_nan.c 
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

/*double tmp = sqrt(-1); */


double myinf = (1.0/0.0);
double not_inf = 1.79769313486231570815E308;
double mynan = (1.0/0. * 0.0);
double not_nan = 0.0;

int main() {

  printf("myinf = %g\tis myinf inf: %d\n", myinf, isinf(myinf));
  printf("not_inf = %g\tis not_inf inf: %d\n", not_inf, isinf(not_inf));

  printf("mynan = %g\tis mynan nan: %d\n", mynan, isnan(mynan));
  printf("not_nan = %g\tis not_nan nan: %d\n", not_nan, isnan(not_nan));

  return 0;
}
```

As you can see, if one uses the builtin ```isinf``` and ```isnan`` to test out a few numbers:

```
devel12-108: gcc tst_inf_nan.c 
devel12-109: a.out
myinf = inf	is myinf inf: 1
not_inf = 1.79769e+308	is not_inf inf: 0
mynan = nan	is mynan nan: 1
not_nan = 0	is not_nan nan: 0
```
QED


There are two problems with the line:
```
double INFINITY = strtod("INF");  /* 99e999; */
```
1) The function ```strtod``` takes two arguments:
```
NAME
       strtod, strtof, strtold - convert ASCII string to floating-point number

SYNOPSIS
       #include <stdlib.h>

       double strtod(const char *nptr, char **endptr);
       float strtof(const char *nptr, char **endptr);
       long double strtold(const char *nptr, char **endptr);
```
2) modern compilers will only allow one to call a function from within a function.  If I comment out the call to the ```sqrt``` function like:

```
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

double tmp = sqrt(-1);


double myinf = (1.0/0.0);
double not_inf = 1.79769313486231570815E308;
double mynan = (1.0/0. * 0.0);
double not_nan = 0.0;

int main() {

  printf("myinf = %g\tis myinf inf: %d\n", myinf, isinf(myinf));
  printf("not_inf = %g\tis not_inf inf: %d\n", not_inf, isinf(not_inf));

  printf("mynan = %g\tis mynan nan: %d\n", mynan, isnan(mynan));
  printf("not_nan = %g\tis not_nan nan: %d\n", not_nan, isnan(not_nan));

  return 0;
}
```
So I can't compile it:

```
devel12-115: gcc tst_inf_nan.c
tst_inf_nan.c:5:1: error: initializer element is not constant
 double tmp = sqrt(-1);
 ^
devel12-116: 
```